### PR TITLE
Fix whitelist/blacklist regex too broad

### PR DESCRIPTION
**Pull request by Gnouc**
Fixes https://github.com/TykTechnologies/tyk/issues/1944

### CODE REVIEW

**Comment by Gnouc** [comment]
cc @buger

**Comment by buger** [comment]
Thank you for the PR! 

I think we can make it a bit smaller, like this:

```
apiLangIDsRegex := regexp.MustCompile(`{([^}]*)}`)
asRegexStr := apiLangIDsRegex.ReplaceAllString(stringSpec, `([^/]*)`)
```

Basically removing logic of detection end of string. And at the same time modifying tests to do that:
```
v.Paths.WhiteList = []string{"/foo/{fooId}$", "/foo/{fooId}/bar/{barId}$"}
```

Basic idea here that here at Tyk we have backward compatibility policy, and this regexp was here for a few years, so there is chance that some of our customers relied on its "confusing" behavior. Thankfuly it was possible to fix it using "$" at the end of string, to explicitly set end of the string. In case of #1944  bug this trick was not working for urls containing {...}. Goal of this PR in this case can be making {...} groups support addign "$" to the end of URL.

Hope it makes sense!

File: api_definition_test.go [hunk 1]
```diff
@@ -133,6 +133,27 @@ func TestWhitelist(t *testing.T) {
...
+			updateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
+				v.Paths.WhiteList = []string{"/foo/{fooId}", "/foo/{fooId}/bar/{barId}"}
+				v.UseExtendedPaths = false
+			})
+
+			spec.Proxy.ListenPath = "/"
+		})
+
+		ts.Run(t, []test.TestCase{
+			{Path: "/foo", Code: 403},
```
Comment (asoorm) [inline api_definition_test.go]: would be nice to use http status code constants.

**Comment by Gnouc** [comment]
@buger @asoorm Updated the PR, please take a look.

**Comment by Gnouc** [comment]
@buger @asoorm any updating?

**Pull request merged**